src/software/thunderscope/play/refereeinfo_widget.py

```diff
@@ -36,15 +35,13 @@ def __init__(self, buffer_size: int = 1) -> None:
 
     def refresh(self) -> None:
         """Update the referee info widget with new referee information"""
-        referee = self.referee_buffer.get(block=False, return_cached=False)
+        referee = self.referee_buffer.get(block=False, return_cached=False).GetOptions()
 
         # Updating QTableWidget could be expensive, so we only update if there is new data
         if referee is None:
             return
 
-        referee_msg_dict = MessageToDict(referee)
-
-        if not referee_msg_dict:
+        if not referee:
             return
 
         stage_time_left_s = (
```

Comment:
you need to update:
`referee_msg_dict["stageTimeLeft"]`
`referee_msg_dict["packetTimestamp"]`